openlibrary/plugins/openlibrary/js/edit.js

```diff
@@ -15,6 +15,125 @@ function update_len() {
+ * @param string removeWorkTitle - the text to show when hovering over the remove button
+ * removeWorkText is a parameter for i18n purposes
+**/
+function render_work_field(i, work, removeWorkTitle){
+    return `<div class="input">
+        <input name="works--${i}" class="work work-autocomplete" type="text" id="work-${i}"
+               value="${work.key.split('/')[-1]}"/>
+        <input name="edition--works--${i}--key" type="hidden" id="work-${i}-key" value="${work.key}"/>
+        <a href="javascript:;" class="remove red plain hidden" title="${removeWorkTitle}">[x]</a>
+    </div>`
+}
+
+/**
+ * Returns the HTML string for one item of the work input field dropdown.
+ * @param object work - has many properties used to show the user key details of the work
+ * @param string newWorkText - the text to show when hovering over a dropdown item
+ * @param string by - the text to show when hovering over a dropdown item
+ * newWorkText and by are parameterized for i18n purposes
+**/
+function render_work_autocomplete_item(work, newWorkText, by){
+    if (work.key === '__new__') {
+        return `<div class="ac_work ac_addnew">
+            <span class="action">${newWorkText}</span>
+        </div>`
+    }
+
+    const coverImg = `<img src="https://covers.openlibrary.org/b/id/${work.cover_i}-M.jpg" alt="Cover of ${work.title}">`
+    const firstYearSpan = `<span class="first_publish_year">(${work.first_publish_year})</span>`
+    const byLine = `<span class="byline">${by} <span class="authors">${work.author_name.join(', ')}</span></span>`
+    return `
+        <div class="ac_work" title="Select this work">
+            <div class="cover">
+                ${work.cover_i ? coverImg : ''}
+            </div>
+            <span class="olid">${work.key.split('/').pop()}</span>
+            <span class="name">
+                <span class="title">${work.full_title}</span>
+                ${work.first_publish_year ? firstYearSpan : ''}
+            </span>
+            ${work.author_name ? byLine : ''}
+            &bull;
+            <span class="edition_count">${work.edition_count} edition${work.edition_count === 1 ? '' : 's'}</span>
+        </div>
+        `
+}
+
+export function initEditionEditPage(){
+    const editionPageData = document.querySelector('[value="edition-edit-page"]');
+    const isPrivilegedUser = editionPageData.dataset.isprivilegeduser === 'true';
+    const newWorkText = editionPageData.dataset.newwork;
+    const removeWorkText = editionPageData.dataset.removework;
+    const selectLanguage = editionPageData.dataset.selectlanguage;
+    const removeLanguage = editionPageData.dataset.removelanguage;
+    const anotherLanguage = editionPageData.dataset.removelanguage;
+    const by = editionPageData.dataset.by;
+    $(function() {
+        ['#languages','#translated_from_languages'].forEach((selector)=>{
+            $(selector).setup_multi_input_autocomplete(
```

Comment:
Might make sense to use document.getElementById since you are using the forEach loop above, otherwise you might consider replacing the forEach with `$('#languages','#translated_from_languages')` which is the same but less lookups.

Reply:
I don't want to dig into why but using `$('#languages','#translated_from_languages')` actually doesn't work. Any query that has multiple results doesn't work. The issue is that if you hit "add language" the second input box won't have autocomplete.